hack/validation/kubelet.sh

```diff
@@ -1,22 +1,25 @@
 # Kubelet Validation 
 
-# The regular expression adds validation for kubelet.kubeReserved and kubelet.systemReserved values of the map are resource.Quantity
-# Quantity: https://github.com/kubernetes/apimachinery/blob/d82afe1e363acae0e8c0953b1bc230d65fdb50e2/pkg/api/resource/quantity.go#L100
-# NodeClaim Validation:
-yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.kubelet.properties.kubeReserved.additionalProperties.pattern = "^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$"' -i pkg/apis/crds/karpenter.sh_nodeclaims.yaml
-yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.kubelet.properties.systemReserved.additionalProperties.pattern = "^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$"' -i pkg/apis/crds/karpenter.sh_nodeclaims.yaml 
+# Adding validation to both v1 and v1beta1 APIs
+# Version = 0 // v1 API 
```

Comment:
nit: Not necessary, but can you just do an enum parser and parse out v1beta1 or v1 from the passed args? That will make it a bit easier to read and I don't think should be a significant increase in the number of lines of code that you have here

Reply:
So I was going back and forth with that idea and I came to the conclusion that for now, since we want to do the same validation parser based approach would not be necessary. I'm planning on switching to a parser if the validation differ between v1 and v1beta1 has a big difference for readability